packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -335,6 +336,13 @@ function createOpenTagMarkup(
     if (propValue == null) {
       continue;
     }
+    // Do not render values for password inputs
+    if (
+      isPasswordInput &&
+      (propKey === 'value' || propKey === 'defaultValue')
+    ) {
+      continue;
+    }
```

Comment:
I don't feel great about this, is there an earlier place I can sift out the value attribute for password inputs?

I think we need to filter out the value attribute on password inputs anyway, even if we eliminate value attribute syncing altogether. The value attribute for most input will still be sent down.